protostar/cairo_testing/execution_environments/cairo_setup_case_execution_environment.py

```diff
@@ -0,0 +1,44 @@
+from typing import cast
+
+from starkware.cairo.lang.compiler.program import Program
+
+from protostar.cairo import HintLocalsDict
+from protostar.cairo_testing.cairo_test_execution_state import CairoTestExecutionState
+from protostar.cairo_testing.cheatcode_factories.cairo_setup_cheatcode_factory import (
+    CairoSetupCheatcodeFactory,
+)
+from protostar.cairo_testing.execution_environments.cairo_execution_environment import (
+    CairoExecutionEnvironment,
+)
+from protostar.cheatable_starknet.cheatable_cached_state import CheatableCachedState
+from protostar.testing.test_context import TestContextHintLocal
+
+
+class CairoSetupCaseExecutionEnvironment(CairoExecutionEnvironment):
+    def __init__(
+        self,
+        program: Program,
+        state: CairoTestExecutionState,
+    ):
+        super().__init__(state, program, self._get_hint_locals(state))
+
+    async def execute(self, function_name: str):
+        with self.state.output_recorder.redirect("setup case"):
+            await self.run_cairo_function(function_name)
+
+    def _get_hint_locals(self, state: CairoTestExecutionState) -> HintLocalsDict:
+        hint_locals: HintLocalsDict = {}
+        cheatcode_factory = CairoSetupCheatcodeFactory(  # For now there is no reason to create another factory file
+            cheatable_state=cast(CheatableCachedState, state.starknet.state.state),
```

Comment:
```suggestion
            cheatable_state=state.cheatable_state,
```
I think this cast should be moved to `CairoTestExecutionState`.

Reply:
That is not possible ATM. It will be, when we pass `CheatableCachedState` in the `TestExecutionState` class instead of `Starknet`. (#1389)

Reply:
```py
    @property
    def cheatable_state(self):
        return cast(CheatableCachedState, self.starknet.state.state)
```

Reply:
Ah right, that one is possible.